From 5413b8825db6eecc6f245854a6bce58e4dee3294 Mon Sep 17 00:00:00 2001
From: "Dr. David Alan Gilbert" <dgilbert@redhat.com>
Date: Tue, 7 Feb 2023 17:57:39 +0000
Subject: [PATCH 20/20] virtio-rng-pci: fix transitional migration compat for
 vectors

RH-Author: Dr. David Alan Gilbert <dgilbert@redhat.com>
RH-MergeRequest: 147: virtio-rng-pci: fix transitional migration compat for vectors
RH-Bugzilla: 2162569
RH-Acked-by: Miroslav Rezanina <mrezanin@redhat.com>
RH-Acked-by: Thomas Huth <thuth@redhat.com>
RH-Acked-by: Gerd Hoffmann <kraxel@redhat.com>
RH-Commit: [1/1] 6e2bd111cd56808fccf2c0464a40f7784fd893a2 (dagrh/c-9-s-qemu-kvm)

In upstream bad9c5a5166/downstream 46e08bafe9ed  I fixed the virito-rng-pci
migration compatibility, but it was discovered that we also need to fix
the other aliases of the device for the transitional cases.

I've sent upstream:
https://lists.gnu.org/archive/html/qemu-devel/2023-02/msg01926.html
but downstream we need to change the downstream machine type anyway,
so it's not quite identical.

Fixes: 9ea02e8f1 ('virtio-rng-pci: Allow setting nvectors, so we can use MSI-X')

Signed-off-by: Dr. David Alan Gilbert <dgilbert@redhat.com>
---
 hw/core/machine.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/hw/core/machine.c b/hw/core/machine.c
index 7adbac6f87..3ee638394b 100644
--- a/hw/core/machine.c
+++ b/hw/core/machine.c
@@ -58,6 +58,9 @@ GlobalProperty hw_compat_rhel_9_1[] = {
   { "virtio-device", "queue_reset", "false" },
   /* hw_compat_rhel_9_1 bz 2155749 */
   { "virtio-rng-pci", "vectors", "0" },
+  /* hw_compat_rhel_9_1 bz 2162569 */
+  { "virtio-rng-pci-transitional", "vectors", "0" },
+  { "virtio-rng-pci-non-transitional", "vectors", "0" },
 };
 const size_t hw_compat_rhel_9_1_len = G_N_ELEMENTS(hw_compat_rhel_9_1);
 
-- 
2.31.1

